### PR TITLE
Add Template Functions & Quota Case

### PR DESCRIPTION
**Pull request by zxiao-redhat**
Hi @kasturinarra, @xingxingxia, @Phaow, I added a few helper functions to work with k8s template (with variables). To showcase the usage of these functions, I automated @wangke19 's quota case. Please take a look if you have time. Thanks!

Tests:
```
passed: (1m24s) 2022-10-10T03:39:05 "[area/quota] Author:zxiao-Critical-[API] Verify that quota works for cluster-scoped resources across all namespaces in the workspace [Suite:kcp/smoke/parallel]"
```

### CODE REVIEW

File: test/extended/quota/quota.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+		k.SetupWorkSpace()
+
+		g.By("# List secrets, config-map resources")
+		err := k.Run("get").Args("cm").Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		err = k.Run("get").Args("secrets").Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("# Create quota under the current workspace")
```
Comment (wangke19) [inline test/extended/quota/quota.go:30]: Currently, only support creating quota in a namespace called admin. Detail see https://polarion.engineering.redhat.com/polarion/redirect/project/OSE/workitem?id=OCP-53980, something wrong with counting worksapce, detail see https://coreos.slack.com/archives/C020QAF9D46/p1664419562486969?thread_ts=1663600223.134799&cid=C020QAF9D46

**Comment by xingxingxia** [comment]
@wangke19 do you have other review comments about the quota feature part's code? If yes, please go on : ) . CC @kasturinarra, topic related: https://url.corp.redhat.com/a58c3d1

**Pull request merged**